ioservice/io_foms.c

```diff
@@ -2357,6 +2357,7 @@ static int m0_io_fom_cob_rw_tick(struct m0_fom *fom)
 					      m0_fom_tx(fom));
 			if (bc_rc != 0)
 				M0_ERR_INFO(bc_rc, "Failed to update cob_size");
+			m0_cob_put(cob);
```

Comment:
I don't think we can call m0_cob_put() here.  cob attributes are being used in  io_finish(), which eventually calling m0_cob_put()

Reply:
This will just free the local instance which was allocated memory in fom_cob_locate() call done few lines above.

Reply:
OK. COB object is being stored in fom object [m0_io_fom_cob_rw->fcrw_cob]. Seems we can free it, as it is not being used elsewhere.
We have the same issue in functions, io_fom_cob2file()  and m0_io_fom_cob_rw_tick().   In both these functions, fom_cob_locate() is being called with allocates new COB object, but is not being freed. Please check once. Thanks.

Reply:
For `io_fom_cob2file()` we have a `m0_cob_put()` call at the end of `io_launch()`.

Reply:
OK